src/CommonITILSatisfaction.php

```diff
@@ -212,6 +215,20 @@ public function prepareInputForUpdate($input)
             }
         }
 
+        if (array_key_exists('satisfaction', $input) && $input['satisfaction'] >= 0) {
+            $item = static::getItemtype();
+            $item = new $item();
+            $fkey = static::getIndexName();
+            if ($item->getFromDB($this->fields[$fkey] ?? $this->fields[$fkey])) {
```

Comment:
```suggestion
            if ($item->getFromDB($input[$fkey] ?? $this->fields[$fkey])) {
```

Reply:
Nice catch, forgot to change to first variable name here.